API/RecommendationService/offline_data_service.py

```diff
@@ -5,26 +5,29 @@
 from .util import get_latest_cmd, RecommendationSource, RecommendType, generated_cosmos_type, CosmosType
 
 
-async def get_recommend_from_offline_data(command_list, recommend_type, error_info, top_num=50):
+async def get_recommend_from_offline_data(command_list, recommend_type, top_num=50):
     loop = asyncio.get_event_loop()
-    cosmos_type = generated_cosmos_type(recommend_type, error_info)
     commands = get_latest_cmd(command_list, 2)
 
-    totalcount_threshold = int(os.environ["Solution_TotalCount_Threshold"]) if cosmos_type == CosmosType.Solution else int(os.environ["Command_TotalCount_Threshold"])
-    ratio_threshold = int(os.environ["Solution_Ratio_Threshold"]) if cosmos_type == CosmosType.Solution else int(os.environ["Command_Ratio_Threshold"])
+    totalcount_threshold = int(os.environ["Command_TotalCount_Threshold"])
+    ratio_threshold = int(os.environ["Command_Ratio_Threshold"])
 
-    if cosmos_type == CosmosType.Solution:
-        return get_recommend_from_cosmos(commands[-1:], recommend_type, error_info, totalcount_threshold, ratio_threshold, top_num)
+    # The recommended content matching the last two commands is preferred. If there is no data, it will fall back to the situation of matching the last command
+    result_2_future = loop.run_in_executor(None, get_recommend_from_cosmos, commands[-2:], recommend_type, None, totalcount_threshold, ratio_threshold, top_num)
+    result_future = loop.run_in_executor(None, get_recommend_from_cosmos, commands[-1:], recommend_type, None, totalcount_threshold, ratio_threshold, top_num)
```

Comment:
Please also use `asyncio.create_task` instead of `loop.run_in_executor`

Reply:
Done